camel/agents/chat_agent.py

```diff
@@ -112,17 +145,65 @@ def __init__(
 
         self.model: ModelType = (model if model is not None else
                                  ModelType.GPT_3_5_TURBO)
-        self.model_config: ChatGPTConfig = model_config or ChatGPTConfig()
         self.message_window_size: Optional[int] = message_window_size
+        self.model_config_dict: dict
```

Comment:
This was because when function calling is enabled, I need to define `self.model_config` as class `FuncConfig`, which is a new dataclass as the configuration including `functions`. As I need to add function objects into this `FuncConfig`, `FuncConfig` cannot be a frozen dataclass which prevents it from inheriting `ChatGPTConfig`. Thus `FuncConfig` will be a completely different class from `ChatGPTConfig`. 
Then the problem comes that `mypy` will report an error if I declare `self.model_config: FuncConfig = ...` and `self.model_config: ChatGPTConfig = ...` in the 2 cases of if-else statement (I remember it will be a duplicate. Thus, considering only the dictionary will essentially be made use by the backend model to call OpenAI API, I just kept this as a field of type `dict` and let it load values from the 2 different dataclasses.